src/driver/mongodb.rs

```diff
@@ -22,7 +22,7 @@ impl Driver for MongoDriver {
   fn connect(uri: &str) -> Result<Self, Error> {
     let config = try!(connstring::parse(uri));
 
-    if let Some(db_name) = config.clone().database {
+    if let Some(db_name) = config.database.clone() {
```

Comment:
yes, but what we pass is an `&str` try just dereferincing the `String` into a `&str` then passing that into the `db` method.